lib/src/runner/configuration.dart

```diff
@@ -605,4 +450,26 @@ class Configuration {
           Map<Object, Configuration> map2) =>
       mergeMaps(map1, map2,
           value: (config1, config2) => config1.merge(config2));
+
+  /// Returns a copy of this [Configuration] with all [chosenPresets] resolved
+  /// against [presets].
+  Configuration _resolvePresets() {
+    if (chosenPresets.isEmpty || presets.isEmpty) return this;
+
+    var newPresets = new Map<String, Configuration>.from(presets);
+    var merged = chosenPresets.fold(empty, (merged, preset) {
+      if (!newPresets.containsKey(preset)) return merged;
```

Comment:
if `null` isn't possible in presets, then you can combine these two operations in one dictionary lookup:
```dart
var newPreset = newPresets.remove(preset);
return newPreset != null ? merged.merge(newPreset) : merged;
```
https://api.dartlang.org/stable/1.20.1/dart-core/Map/remove.html

whether it's worth a small tweak like that, not sure. :)